CoreScriptsRoot/Modules/Settings/SettingsHub.lua

```diff
@@ -940,12 +941,30 @@ local function CreateSettingsHub()
 	end
 
 	function this:ShowShield()
-		this.Shield.BackgroundTransparency = SETTINGS_SHIELD_TRANSPARENCY
+		this.Shield.BackgroundTransparency = VREnabled and SETTINGS_SHIELD_VR_TRANSPARENCY or SETTINGS_SHIELD_TRANSPARENCY
 	end
 	function this:HideShield()
 		this.Shield.BackgroundTransparency = 1
 	end
 
+	local function enableVR()
+		this.Shield.Parent.ClipsDescendants = false
+		this:HideShield()
+	end
+
+	local UserInputService = game:GetService("UserInputService")
+	local UISChanged
+	local function OnVREnabled(prop)
+		if prop == "VREnabled" and UserInputService.VREnabled then
+			VREnabled = true
+			enableVR()
+			UISChanged:disconnect()
```

Comment:
I like to set `UISChanged = nil` after disconnecting to free up references. You may also want to wrap it in a nil check too then